rand_core/src/lib.rs

```diff
@@ -208,6 +208,20 @@ pub trait RngCore {
 /// [`BlockRngCore`]: block::BlockRngCore
 pub trait CryptoRng {}
 
+/// An extension trait to support trait objects that implement [`RngCore`] and
+/// [`CryptoRng`]. Upcasting to [`RngCore`] is supported via the
+/// [`CryptoRngCore::upcast`] method.
```

Comment:
What would be a good example we could add here?

Reply:
We should say that `CryptoRngCore` is automatically implemented for `T: CryptoRng + RngCore` and direct implementations are not recommended, but that the trait may be useful as a bound. So...

```rust
/// CryptoRng + RngCore
///
/// This trait is automatically implemented for any `T: CryptoRng + RngCore`, and may be used as a trait object, e.g.
/// ```
/// fn make_token(rng: &mut CryptoRngCore) -> [u8; 32] {
///     let mut buf = [0u8; 32];
///     rng.fill(&mut buf);
///     buf
/// }
/// ```
```

Reply:
> direct implementations are not recommended

So we plan to keep `CryptoRng` instead of replacing it with `CryptoRngCore`? I'm fine with that, because
1. it's backwards compatible, and
2. it avoids the boilerplate code for CSPRNG implementations.

Reply:
I don't want to reread that *entire* thread, but I think that's what we concluded?

Reply:
I don't think we made a decision about deprecating/removing `CryptoRng`, but we might as well make it here.